File: Assets/Scripts/Matching/MenuMusic.js
#pragma strict
var lesTunes: AudioClip[];
private var t: float = 0.0;
function Start ()
{
	GetComponent.<AudioSource>().clip = lesTunes[Random.Range(0, lesTunes.length)];
	GetComponent.<AudioSource>().Play();
}

function Update ()
{
	t += Time.deltaTime;
	if (t>= GetComponent.<AudioSource>().clip.length)
	{
		GetComponent.<AudioSource>().Stop();
		GetComponent.<AudioSource>().clip = lesTunes[Random.Range(0, lesTunes.length)];
		GetComponent.<AudioSource>().Play();
		t = 0.0;
	}
}

File: Assets/Scripts/CSharp/MenuMusic.cs
using UnityEngine;
[RequireComponent(typeof(AudioSource))]
public class MenuMusic : MonoBehaviour
{
    private float t = 0;
    private AudioSource myAudio;
    public AudioClip[] lesTunes;

	void Start ()
    {
        myAudio = GetComponent<AudioSource>();
        myAudio.clip = lesTunes[Random.Range(0, lesTunes.Length)];
        myAudio.Play();
    }
	
	void Update ()
    {
        t += Time.deltaTime;
        if (t >= myAudio.clip.length)
        {
            myAudio.Stop();
            myAudio.clip = lesTunes[Random.Range(0, lesTunes.Length)];
            myAudio.Play();
            t = 0;
        }
    }
}
